### PR TITLE
Remove superfluous phpdoc tags

### PR DESCRIPTION
**Pull request by tigitz**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no 
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | ...
| License       | MIT

As Symfony is moving towards type hinted code, this PR is attempting to remove the now superfluous annotations

See #33150 to make this cleaning as an automatic process

### CODE REVIEW

File: src/Symfony/Bridge/Doctrine/Tests/Fixtures/BaseUser.php [hunk 1]
```diff
@@ -22,8 +22,6 @@ class BaseUser
     /**
      * BaseUser constructor.
      *
```
Comment (nicolas-grekas) [inline src/Symfony/Bridge/Doctrine/Tests/Fixtures/BaseUser.php]: the fixer should also remove this line if possible

File: src/Symfony/Component/Console/Question/Question.php [hunk 2]
```diff
@@ -188,7 +188,6 @@ public function setAutocompleterCallback(callable $callback = null): self
     /**
      * Sets a validator for the question.
      *
-     * @param callable|null $validator
      *
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Console/Question/Question.php]: can extra blank lines be removed by the tool too?

File: src/Symfony/Component/Mime/Test/Constraint/EmailHtmlBodyContains.php [hunk 3]
```diff
@@ -44,11 +44,6 @@ protected function matches($message): bool
         return false !== mb_strpos($message->getHtmlBody(), $this->expectedText);
     }
 
-    /**
-     * @param RawMessage $message
-     *
-     * {@inheritdoc}
-     */
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Mime/Test/Constraint/EmailHtmlBodyContains.php]: why superfuous here?

**Review by nicolas-grekas — COMMENTED** [review]
"mixed" would not be considered superfluous to me (to answer your question on the issue)

File: src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php [hunk 4]
```diff
@@ -309,11 +307,8 @@ protected function prepareForDenormalization($data)
      * Returns the method to use to construct an object. This method must be either
      * the object constructor or static.
      *
-     * @param array            $data
-     * @param string           $class
-     * @param array            $context
-     * @param \ReflectionClass $reflectionClass
-     * @param array|bool       $allowedAttributes
+     * @param string     $class
```
Comment (tigitz) [inline src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php]: this should probably added as a type hint directly now, should I do it in the scope of this cleaning ?

File: src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php [hunk 5]
```diff
@@ -330,12 +325,8 @@ protected function getConstructor(array &$data, $class, array &$context, \Reflec
      * is removed from the context before being returned to avoid side effects
      * when recursively normalizing an object graph.
      *
-     * @param array            $data
-     * @param string           $class
-     * @param array            $context
-     * @param \ReflectionClass $reflectionClass
-     * @param array|bool       $allowedAttributes
-     * @param string|null      $format
+     * @param string     $class
```
Comment (tigitz) [inline src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php]: Same https://github.com/symfony/symfony/pull/33151/files#r313605371

File: src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php [hunk 6]
```diff
@@ -285,10 +281,7 @@ protected function getAttributes($object, string $format = null, array $context)
     /**
      * Gets the attribute value.
      *
-     * @param object      $object
-     * @param string      $attribute
-     * @param string|null $format
-     * @param array       $context
+     * @param object $object
```
Comment (tigitz) [inline src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php]: Same as https://github.com/symfony/symfony/pull/33151/files#r313605371

File: src/Symfony/Component/Mime/Test/Constraint/EmailHtmlBodyContains.php (hunk 3 above)
Comment (tigitz) [reply src/Symfony/Component/Mime/Test/Constraint/EmailHtmlBodyContains.php]: I thought it was a typo as the annotated param here is `$message`, probably copy pasted from the function above, while the real param is `$email` here but with no indications on it's type it seems. So I decided to simply remove it.

I'll keep the inheritdoc though.

File: src/Symfony/Component/Console/Question/Question.php (hunk 2 above)
Comment (tigitz) [reply src/Symfony/Component/Console/Question/Question.php]: hmm shouldn't fabbot.io catch those ?

I ran php-cs-fixer with only `'no_superfluous_phpdoc_tags' => ['allow_mixed' => true],` as a rule to speed up things, so others rules from `.php_cs.dist` might handle these.

What's clear is that the `no_superfluous_phpdoc_tags` doesn't handle it itself after it's cleaning

I'll check my guess and see if there's even another rule we can add that can do this check if not in `.php_cs.dist` file already

**Comment by tigitz** [comment]
Cleaning of some files went a little bit too far and broke some php doc extraction tests :sweat_smile:, will fix this

**Review by lyrixx — APPROVED** [review]
:+1: for the workflow part

File: src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php (hunk 4 above)
Comment (derrabus) [reply src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php]: I'll add the type declaration with #33154.

btw, not all of the parameters are actually in use, as far as I can tell.

**Comment by nicolas-grekas** [comment]
Thank you @tigitz 
I've also added the `phpdoc_trim_consecutive_blank_line_separation` rule and opened https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/4504

File: src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php (hunk 4 above)
Comment (tigitz) [reply src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php]: Ok, noted, thanks, resolving conversation

**Comment by nicolas-grekas** [comment]
This is now merged up to master :tada:

**Pull request merged**